### PR TITLE
add scikit-learn feature selection to model validation 

### PR DESCRIPTION
**Pull request by qiagu**
1) add feature selection function in macro
2) introduce feature selection UI and function in model validation tool
3) add GridSearchCV 
4) do feature selection and cross validation in pipeline

Credit to @jgoecks

### CODE REVIEW

File: tools/sklearn/main_macros.xml [hunk 1]
```diff
@@ -975,8 +1023,8 @@ def read_columns(f, c, return_df=False, **args):
     <param argument="scoring" type="text" value="" optional="true" label="scoring" help="A metric used to evaluate the estimator"/>
   </xml>
 
-  <xml name="pre_dispatch">
-    <param argument="pre_dispatch" type="text" value="all" optional="true" label="pre_dispatch" help="Number of predispatched jobs for parallel execution"/>
+  <xml name="pre_dispatch" token_type="text" token_default_value="all" token_help="Number of predispatched jobs for parallel execution">
+    <param argument="pre_dispatch" type="@TYPE@" value="@DEFAULT_VALUE@" optional="true" label="pre_dispatch" help="@HELP@"/>
```
Comment (bgruening) [inline tools/sklearn/main_macros.xml]: !! Great feature isn't it? :)

File: tools/sklearn/model_validation.xml [hunk 2]
```diff
@@ -51,63 +55,119 @@ y = read_columns(
 )
 y=y.ravel()
 
-validator = params["model_validation_functions"]["selected_function"]
-validator = getattr(sklearn.model_selection, validator)
 options = params["model_validation_functions"]["options"]
 if 'scoring' in options and options['scoring'] == '':
     options['scoring'] = None
+if 'pre_dispatch' in options and options['pre_dispatch'] == '':
```
Comment (bgruening) [inline tools/sklearn/model_validation.xml]: `if options.get('pre_dispatch', False):`

File: tools/sklearn/main_macros.xml (hunk 1 above)
Comment (qiagu) [reply tools/sklearn/main_macros.xml]: Yes, it is.

File: tools/sklearn/model_validation.xml (hunk 2 above)
Comment (qiagu) [reply tools/sklearn/model_validation.xml]: A little tricky here. I'll be happy if you can make it better.

**Comment by qiagu** [comment]
Thanks for merging!

**Pull request merged**